### PR TITLE
Fix serde for ArrayOfDoublesSketchConstantPostAggregator.

### LINKED ISSUE #16539: Issue with PostAggregator arrayOfDoublesSketchConstant in latest Druid 29.0.1
Getting Exception when using PostAggregator `arrayOfDoublesSketchConstant` in the groupby Query. 

### Affected Version

Currently running, Druid version 25.0.0 with custom jar for Datasketches, trying to use the latest druid with postaggregator `arrayOfDoublesSketchConstant`

### Description

Getting an exception : 
```
{
    "error": "Unknown exception",
    "errorMessage": "Unhandled exception made it to the top",
    "errorClass": "com.fasterxml.jackson.databind.exc.ValueInstantiationException",
    "host": "X.X.X.X"
}
```

Example Query 
```
{
    "dataSource": {
        "type": "table",
        "name": "brq"
    },
    "context": {
        "queryId": "4d516332-92f3-4f6b-a19f-760f0865653d",
        "timeout": 120000
    },
    "queryType": "groupBy",
    "intervals": [
        "2024-05-23T00:00:00.000Z/2024-06-02T00:00:00.000Z"
    ],
    "granularity": "hour",
    "filter": {
        "type": "and",
        "fields": [
            {
                "type": "and",
                "fields": [
                    {
                        "type": "selector",
                        "dimension": "exc",
                        "value": 6
                    },
                    {
                        "type": "and",
                        "fields": [
                            {
                                "type": "selector",
                                "dimension": "country",
                                "value": "US"
                            }
                ]
            },
            {
                "type": "in",
                "dimension": "types",
                "values": [
                    "5"
                ]
            }
        ]
    },
    "aggregations": [
        {
            "type": "arrayOfDoublesSketch",
            "name": "tuple_sketch_agg",
            "fieldName": "tuple_sketch_count"
        }
    ],
    "postAggregations": [
        {
            "type": "arrayOfDoublesSketchToMetricsSumEstimate",
            "name": "psids_tuple_sketch_metric_estimate",
            "field": {
                "type": "arrayOfDoublesSketchSetOp",
                "name": "tuple_sketch_set_ops",
                "operation": "INTERSECT",
                "fields": [
                    {
                        "type": "fieldAccess",
                        "name": "tuple_sketch_post_agg",
                        "fieldName": "tuple_sketch_agg"
                    },
                    {
                        "type": "arrayOfDoublesSketchConstant",
                        "name": "tuple_sketch_const_post_agg",
                        "value": "AQEJAwgBzJP/////////fyMAAAAAAAAAjBCbsE9M6x0OgYV+hyEhMxPw94jlVLRXGyO4rtPzLWYcnHztRP9Wd58FyDQEmRw2ofL5I83Ee3Ik9kmVt7h8c03WBgoFokEcpycf1NYBADgwuTxsghD+KTEcUTzHx95aPNULnBE3Jza9Nop6/YRldD5W5wISgod3wGaHm9fZaBDFTp22xM8mI0jxq5WQp6RBSnjyU1z9RShIAeGt6GwKClGIUp5AqJVK1MFkXNeS2B5ZznJuo/qJN+JPO+Tn1vtp4+u6RbBYliZiAfvzgBPFb2lOrcJkcM5Z612tuz97oHtt+2A/19TsMe/Z9PqIkIx5cSdC+RGmeSj0gehVsshlW/aioTu5KRY5+uSCBGkhFW7KrLuqHsgydwAAAAAAAPA/AAAAAAAAQ0AAAAAAAADwPwAAAAAAAPA/AAAAAAAA8D8AAAAAAAAAQAAAAAAAAABAAAAAAAAA8D8AAAAAAADwPwAAAAAAAPA/AAAAAAAA8D8AAAAAAAAIQAAAAAAAAPA/AAAAAAAA8D8AAAAAAADwPwAAAAAAAPA/AAAAAAAA8D8AAAAAAADwPwAAAAAAAPA/AAAAAAAA8D8AAAAAAADwPwAAAAAAAPA/AAAAAAAAAEAAAAAAAADwPwAAAAAAAPA/AAAAAAAA8D8AAAAAAADwPwAAAAAAAPA/AAAAAAAA8D8AAAAAAADwPwAAAAAAAPA/AAAAAAAA8D8AAAAAAADwPwAAAAAAAAhAAAAAAAAACEA="
                    }
                ],
            }
        }
    ],
    "dimensions": []
}
```

`tuple_sketch_count` is a column which is of type ArrayofDoubleSketch

### PR DESCRIPTION
**Pull request by gianm**
The version originally added in #13819 was missing an annotation for the "value" property. Fixes #16539.

Line endings for ArrayOfDoublesSketchConstantPostAggregator.java are changed from `\r\n` to `\n`. Try viewing with whitespace ignored to make the diff easier to read. The really meaningful change is adding a single line.

Adds a serde test, and improves various other datasketches post-aggregator serde tests to deserialize into PostAggregator. This verifies that the type information is set up correctly.

### CODE REVIEW

**Review by cryptoe — APPROVED** [review]
The diff was hard to see. Thanks for the tip.

**Review by kgyrtkirk — APPROVED** [review]

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchUnionPostAggregatorTest.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import org.apache.datasketches.hll.TgtHllType;
 import org.apache.druid.jackson.DefaultObjectMapper;
 import org.apache.druid.query.aggregation.PostAggregator;
+import org.apache.druid.query.aggregation.datasketches.kll.KllSketchModule;
```
Comment (asdf2014) [inline extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchUnionPostAggregatorTest.java:27]:
```suggestion
```

**Review by asdf2014 — APPROVED** [review]
Overall LGTM, still need remove this import as it never has been used to pass the style check

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToCDFPostAggregatorTest.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 import org.apache.druid.query.aggregation.Aggregator;
 import org.apache.druid.query.aggregation.PostAggregator;
 import org.apache.druid.query.aggregation.TestDoubleColumnSelectorImpl;
+import org.apache.druid.query.aggregation.datasketches.tuple.ArrayOfDoublesSketchModule;
```
Comment (asdf2014) [inline extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToCDFPostAggregatorTest.java:31]:
```suggestion
```

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToHistogramPostAggregatorTest.java [hunk 3]
```diff
@@ -28,6 +28,7 @@
 import org.apache.druid.query.aggregation.Aggregator;
 import org.apache.druid.query.aggregation.PostAggregator;
 import org.apache.druid.query.aggregation.TestDoubleColumnSelectorImpl;
+import org.apache.druid.query.aggregation.datasketches.tuple.ArrayOfDoublesSketchModule;
```
Comment (asdf2014) [inline extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToHistogramPostAggregatorTest.java:31]:
```suggestion
```

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToQuantilePostAggregatorTest.java [hunk 4]
```diff
@@ -23,6 +23,7 @@
 import nl.jqno.equalsverifier.EqualsVerifier;
 import org.apache.druid.jackson.DefaultObjectMapper;
 import org.apache.druid.query.aggregation.PostAggregator;
+import org.apache.druid.query.aggregation.datasketches.tuple.ArrayOfDoublesSketchModule;
```
Comment (asdf2014) [inline extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToQuantilePostAggregatorTest.java:26]:
```suggestion
```

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToQuantilesPostAggregatorTest.java [hunk 5]
```diff
@@ -28,6 +28,7 @@
 import org.apache.druid.query.aggregation.Aggregator;
 import org.apache.druid.query.aggregation.PostAggregator;
 import org.apache.druid.query.aggregation.TestDoubleColumnSelectorImpl;
+import org.apache.druid.query.aggregation.datasketches.tuple.ArrayOfDoublesSketchModule;
```
Comment (asdf2014) [inline extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToQuantilesPostAggregatorTest.java:31]:
```suggestion
```

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToRankPostAggregatorTest.java [hunk 6]
```diff
@@ -27,6 +27,7 @@
 import org.apache.druid.query.aggregation.Aggregator;
 import org.apache.druid.query.aggregation.PostAggregator;
 import org.apache.druid.query.aggregation.TestDoubleColumnSelectorImpl;
+import org.apache.druid.query.aggregation.datasketches.tuple.ArrayOfDoublesSketchModule;
```
Comment (asdf2014) [inline extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToRankPostAggregatorTest.java:30]:
```suggestion
```

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToStringPostAggregatorTest.java [hunk 7]
```diff
@@ -24,6 +24,7 @@
 import org.apache.druid.jackson.DefaultObjectMapper;
 import org.apache.druid.java.util.common.IAE;
 import org.apache.druid.query.aggregation.PostAggregator;
+import org.apache.druid.query.aggregation.datasketches.tuple.ArrayOfDoublesSketchModule;
```
Comment (asdf2014) [inline extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToStringPostAggregatorTest.java:27]:
```suggestion
```

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/kll/KllFloatsSketchToStringPostAggregatorTest.java [hunk 8]
```diff
@@ -24,6 +24,7 @@
 import org.apache.druid.jackson.DefaultObjectMapper;
 import org.apache.druid.java.util.common.IAE;
 import org.apache.druid.query.aggregation.PostAggregator;
+import org.apache.druid.query.aggregation.datasketches.theta.SketchModule;
```
Comment (asdf2014) [inline extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/kll/KllFloatsSketchToStringPostAggregatorTest.java:27]:
```suggestion
```

**Review by asdf2014 — COMMENTED** [review]
Same as above

**Comment by gianm** [comment]
Fixed the excessive imports. Thanks for reviewing!

File: extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchConstantPostAggregatorTest.java [hunk 9]
```diff
@@ -66,6 +67,24 @@ public void testComparator()
...
+    final PostAggregator there = new ArrayOfDoublesSketchConstantPostAggregator(
+        "p",
+        "AQEJAwgBzJP/////////fwIAAAAAAAAAzT6NGdX0aWUOJvS5EIhpLwAAAAAAAAAAAAAAAAAAAAA="
+    );
+    DefaultObjectMapper mapper = new DefaultObjectMapper();
+    mapper.registerModules(new ArrayOfDoublesSketchModule().getJacksonModules());
+    PostAggregator andBackAgain = mapper.readValue(
+        mapper.writeValueAsString(there),
+        PostAggregator.class
+    );
```
[the remark this answers is not in the document]
Comment (gianm) [reply extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchConstantPostAggregatorTest.java:82]: Perhaps we could use some approach like `EqualsVerifier`. Whatever magic it does to verify `equals` and `hashCode` could possible be used to verify serde as well? I never looked inside it to see what it's doing.

**Pull request merged**